### PR TITLE
popup: close popup on escape press.

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Review by ogoffart — COMMENTED** [review]
I'd put the esc handler after the key_event handler (together with the Tab/Backtab handling a few lines down.

Can you please extend the test in https://github.com/slint-ui/slint/blob/master/tests/cases/elements/popupwindow_close_policy.slint to make sure esc close the popup when it should, and doesn't close whan it shouldn't

File: internal/common/enums.rs [hunk 1]
```diff
@@ -434,13 +434,13 @@ macro_rules! for_each_enums {
 
             // This enum describes the close behaviour of [`PopupWindow`](elements.md#popupwindow)
             enum PopupClosePolicy {
-                /// Closes the `PopupWindow` when user clicks.
+                /// Closes the `PopupWindow` when user clicks or presses the escape key.
                 CloseOnClick,
 
-                /// Closed the `PopupWindow` when user clicks outside of the popup.
+                /// Closed the `PopupWindow` when user clicks outside of the popup or presses the escape key.
```
Comment (DataTriny) [inline internal/common/enums.rs:440]:
```suggestion
                /// Closes the `PopupWindow` when user clicks outside of the popup or presses the escape key.
```

File: tests/cases/elements/popupwindow_close_policy.slint [hunk 2]
```diff
@@ -259,6 +259,21 @@ assert_eq!(instance.get_click_count(), 2);
...
+instance.set_popup_selector(3);
+instance.set_click_count(0);
+slint_testing::send_mouse_click(&instance, 5., 5.);
+assert_eq!(instance.get_click_count(), 1);
+
+// click inside
+slint_testing::send_mouse_click(&instance, 15., 15.);
+assert_eq!(instance.get_popup_clicked(), 2005);
+
+// close by esc
```
Comment (ogoffart) [inline tests/cases/elements/popupwindow_close_policy.slint:272]: Please also add a esc press around line 216 (`Click outside, nothing happens`) to make sure that clicking on esc does not close that popup.


And maybe also extending popupwindow_nested.slint to test that esc closes one and only one popup.

**Review by ogoffart — APPROVED** [review]
Looks good to me, but the test coverage could be improved (see comment)

**Review by ogoffart — APPROVED** [review]

**Pull request merged**